flocker/apiclient/_client.py

```diff
@@ -25,14 +25,14 @@ class Dataset(PClass):
     A dataset in the configuration.
 
     :attr UUID primary: The node where the dataset should manifest.
-    :attr int maximum_size: Size of new dataset, in bytes.
+    :attr maximum_size: Size of new dataset in bytes (as ``int``) or ``None``.
     :attr UUID dataset_id: The UUID of the dataset.
     :attr metadata: A mapping between unicode keys and values.
     :attr bool deleted: If true indicates this dataset should be deleted.
     """
     dataset_id = field(type=UUID, mandatory=True)
     primary = field(type=UUID, mandatory=True)
-    maximum_size = field(type=int, mandatory=True)
+    maximum_size = field(type=(int, None.__class__), mandatory=True)
```

Comment:
:question: Use `type(None)` instead of `None.__class__` here and in other places below.